code/modules/research/xenobiology/crossbreeding/warping

```diff
@@ -0,0 +1,796 @@
+/obj/effect/warped_rune/attackby(obj/item/stack/BC, mob/user)
+	if(!istype(BC,/obj/item/stack/sheet/bluespace_crystal) && !istype(BC,/obj/item/stack/ore/bluespace_crystal))
+		return
+	if(do_after(user, storing_time,target = src)) //the time it takes to nullify it depends on the rune too
+		to_chat(user, "<span class='notice'>You nullify the effects of the rune with the bluespace crystal!</span>")
+		qdel(src)
+		BC.amount--
+		playsound(src, 'sound/effects/phasein.ogg', 20, TRUE)
+		if(BC.amount <= 0)
+			qdel(BC)
+///destroyable by xenomorphs			
+/obj/effect/warped_rune/acid_act() 
+	visible_message("<span class='warning'>The rune is being disolved!</span>")
+	playsound(src, 'sound/items/welder.ogg', 150, TRUE)
+	qdel(src)
+
+//nearly all runes use their turf in some way so we set T to their turf automatically.
+/obj/effect/warped_rune/Initialize()
+	..()
+	T = get_turf(src)
+	RegisterSignal(T, COMSIG_COMPONENT_CLEAN_ACT, .proc/clean_rune)
+	cooldown = world.time + max_cooldown //rune starts on cooldown
+	
+/obj/effect/warped_rune/proc/clean_rune()
+	qdel(src)
+
+ ///using the extract on the floor will "draw" the rune.
+/obj/item/slimecross/warping/afterattack(turf/target, mob/user, proximity)
+	if(!proximity)
+		return
+
+	if(istype(target) && locate(/obj/effect/warped_rune) in target) //check if the target is a floor and if there's a rune on said floor
+		to_chat(user, "<span class='warning'>There is already a bluespace rune here!</span>")
+		return
+	if(istype(target, runepath))       //checks if the target is a rune and then if you can store it
+		if(warp_charge == max_charge)
+			to_chat(user, "<span class='warning'>Your crossbreed is already full!</span>")
+			return
+		if(do_after(user, storing_time,target = target) && warp_charge < max_charge)
+			to_chat(user, "<span class='notice'>You store the rune in your bluespace crossbreed</span>")
+			qdel(target)
+			warp_charge++
+			return
+
+	if(!istype(target) || isspaceturf(target))
+		return
+
+	if(locate(/turf/closed) in target || locate(/obj/structure) in target) // check if there's a wall or a structure in the way
+		to_chat(user, "<span class='warning'>something is in the way of the rune!</span>")
+		return
+
+	if(warp_charge > 0 && do_after(user, drawing_time,target = target)) //finally spawns the right rune if you have charge(s) left
+		playsound(target, 'sound/effects/slosh.ogg', 20, TRUE)
+
+		if(warp_charge == 0) //this is only here to fix a bug where the user can draw multiple runes at once if they spam click
+			to_chat(user, "<span class='warning'>Your crossbreed is empty!</span>")
+			return
+		warp_charge--
+		new runepath(target)
+		to_chat(user, "<span class='notice'>You carefully draw the rune with your crossbreed.</span>")
+
+
+/obj/item/slimecross/warping/grey
+	colour = "grey"
+	effect_desc = "Creates a rune.Extracts the rune is on are absorbed, 8 extracts produces a slime of that color."
```

Comment:
forgot a space after that period

Reply:
just say "Extracts that are on the rune are absorbed" instead of "Extracts the rune is on are absorbed"